coderadar-test/src/test/java/io/reflectoring/coderadar/rest/authentication/AuthenticationServiceIntegrationTest.java

```diff
@@ -0,0 +1,133 @@
+package io.reflectoring.coderadar.rest.authentication;
+
+import io.reflectoring.coderadar.CoderadarConfigurationProperties;
+import io.reflectoring.coderadar.graph.projectadministration.domain.ProjectEntity;
+import io.reflectoring.coderadar.graph.projectadministration.project.repository.ProjectRepository;
+import io.reflectoring.coderadar.graph.useradministration.domain.UserEntity;
+import io.reflectoring.coderadar.graph.useradministration.repository.UserRepository;
+import io.reflectoring.coderadar.rest.ControllerTestTemplate;
+import io.reflectoring.coderadar.useradministration.domain.ProjectRole;
+import io.reflectoring.coderadar.useradministration.port.driver.login.LoginUserCommand;
+import io.reflectoring.coderadar.useradministration.service.UserUnauthenticatedException;
+import io.reflectoring.coderadar.useradministration.service.UserUnauthorizedException;
+import io.reflectoring.coderadar.useradministration.service.login.LoginUserService;
+import io.reflectoring.coderadar.useradministration.service.permissions.SetUserRoleForProjectService;
+import io.reflectoring.coderadar.useradministration.service.security.AuthenticationService;
+import io.reflectoring.coderadar.useradministration.service.security.PasswordUtil;
+import java.util.Date;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+class AuthenticationServiceIntegrationTest extends ControllerTestTemplate {
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private ProjectRepository projectRepository;
+
+  @Autowired private AuthenticationService authenticationService;
+
+  @Autowired private SetUserRoleForProjectService setUserRoleForProjectService;
+
+  @Autowired private LoginUserService loginUserService;
+
+  @Autowired private CoderadarConfigurationProperties coderadarConfigurationProperties;
+
+  private ProjectEntity testProject;
+  private UserEntity testUser;
+
+  private Long projectId;
+
+  @BeforeEach
+  void setUp() {
+    coderadarConfigurationProperties.setAuthentication(
```

Comment:
As far as I know, til now, this is the only test using such a configuration. if there are more configurations like this, we should use a spring profile for tests to handle these configurations for us.

Reply:
I'm not sure what you mean by configuration. The credentials are basically mocked in these tests.

Reply:
I thought this line would basically just set the property `coderadar.authentication.enabled=true` of he `local.application.properties` file. This was a misunderstanding by me.

Reply:
That is correct, however I did not undestand the issue.